drivers/adc/adc_nrfx_saadc.c

```diff
@@ -432,11 +439,37 @@ static int check_buffer_size(const struct adc_sequence *sequence,
 	return 0;
 }
 
+static bool has_single_ended(const struct adc_sequence *sequence)
+{
+	return sequence->channels & m_data.single_ended_channels;
+}
+
+static void correct_single_ended(const struct adc_sequence *sequence)
+{
+	uint16_t channel_id = BIT(0);
```

Comment:
This name may be a bit confusing as the value here is something different from `.channel_id` in `struct adc_channel_cfg` and `channel_id` used in `start_read()`.

Reply:
I see, did think the name was odd, I just reused it because of `BIT(channel_id)` without realizing the `BIT()` part of it :D